src/Nest/Modules/SnapshotAndRestore/Repositories/S3Repository.cs

```diff
@@ -7,131 +7,264 @@ public interface IS3Repository : IRepository<IS3RepositorySettings> { }
 
 	public class S3Repository : IS3Repository
 	{
-		public S3Repository(S3RepositorySettings settings)
-		{
-			Settings = settings;
-		}
+		public S3Repository(IS3RepositorySettings settings) => Settings = settings;
 
 		public IS3RepositorySettings Settings { get; set; }
 		public string Type { get; } = "s3";
 	}
 
 	public interface IS3RepositorySettings : IRepositorySettings
 	{
+		/// <summary>
+		/// The name of the bucket to be used for snapshots. This field is required
+		/// </summary>
 		[JsonProperty("bucket")]
 		string Bucket { get; set; }
 
+		/// <summary>
+		/// The region where bucket is located. Defaults to US Standard.
+		/// </summary>
 		[Obsolete("Removed in NEST 6.x.")]
 		[JsonProperty("region")]
 		string Region { get; set; }
 
+		/// <summary>
+		/// The endpoint to the S3 API. Defaults to AWS’s default S3 endpoint. Note that setting a region overrides the endpoint setting.
+		/// </summary>
+		[Obsolete("Removed in NEST 6.x.")]
+		[JsonProperty("endpoint")]
+		string Endpoint { get; set; }
+
+		/// <summary>
+		/// The protocol to use (http or https). Defaults to value of cloud.aws.protocol or cloud.aws.s3.protocol.
+		/// </summary>
+		[Obsolete("Removed in NEST 6.x.")]
+		[JsonProperty("protocol")]
+		string Protocol { get; set; }
+
+		/// <summary>
+		/// Specifies the path within bucket to repository data.
+		/// Defaults to value of repositories.s3.base_path or to root directory if not set.
+		/// </summary>
 		[JsonProperty("base_path")]
 		string BasePath { get; set; }
 
+		/// <summary>
+		/// The access key to use for authentication. Defaults to value of cloud.aws.access_key.
+		/// </summary>
+		[Obsolete("Removed in NEST 6.x. Sensitive settings must be stored in the elasticsearch keystore. See https://www.elastic.co/guide/en/elasticsearch/plugins/current/repository-s3-client.html")]
 		[JsonProperty("access_key")]
 		string AccessKey { get; set; }
 
+		/// <summary>
+		/// The secret key to use for authentication. Defaults to value of cloud.aws.secret_key.
+		/// </summary>
+		[Obsolete("Removed in NEST 6.x. Sensitive settings must be stored in the elasticsearch keystore. See https://www.elastic.co/guide/en/elasticsearch/plugins/current/repository-s3-client.html")]
 		[JsonProperty("secret_key")]
 		string SecretKey { get; set; }
 
+		/// <summary>
+		/// Big files can be broken down into chunks during snapshotting if needed.
+		/// The chunk size can be specified in bytes or by using size value notation,
+		/// i.e. 1gb, 10mb, 5kb. Defaults to 1gb.
+		/// </summary>
+		[JsonProperty("chunk_size")]
+		string ChunkSize { get; set; }
+
+		/// <summary>
+		/// When set to true metadata files are stored in compressed format.
+		/// This setting doesn't affect index files that are already compressed by default.
+		/// Defaults to false.
+		/// </summary>
 		[JsonProperty("compress")]
 		bool? Compress { get; set; }
 
-		[JsonProperty("concurrent_streams")]
-		int? ConcurrentStreams { get; set; }
```

Comment:
Can't be removed: Binary breaking change

Reply:
Will revert.